test/controllers/NCTSMessageControllerSpec.scala

```diff
@@ -42,275 +38,122 @@ import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
-  private val testBox   = Box(BoxId(testBoxId), Constants.BoxName)
-
-  private val arrivalWithBox = arrivalWithoutBox.copy(notificationBox = Some(testBox))
-
-  private val invalidXml = <test></test>
-
-  private val dateTime = LocalDateTime.now
-
-  private val xml =
-    (<CC007A>
-      <DatOfPreMES9>
-        {Format.dateFormatted(dateTime)}
-      </DatOfPreMES9>
-      <TimOfPreMES10>
-        {Format.timeFormatted(dateTime.toLocalTime)}
-      </TimOfPreMES10>
-    </CC007A>).map(trim)
+  private val messageSender = MessageSender(ArrivalId(1), 1)
 
   override def beforeEach: Unit = {
     super.beforeEach()
-    reset(mockSaveMessageService)
-    reset(mockPushPullNotificationService)
-    reset(mockInboundRequestService)
+    reset(mockMovementMessageOrchestratorService)
   }
 
   "post" - {
 
-    "must return OK, when the service validates and save the message" in {
+    "must return OK" in {
 
-      val message = Arbitrary.arbitrary[MovementMessageWithoutStatus].sample.value
+      val arrival = Arbitrary.arbitrary[Arrival].sample.value
 
-      when(mockInboundRequestService.makeInboundRequest(any(), any(), any())(any()))
-        .thenReturn(Future.successful(Right(InboundMessageRequest(arrivalWithoutBox, GoodsReleased, GoodsReleasedResponse, message))))
+      val message = Arbitrary.arbitrary[MovementMessageWithoutStatus].sample.value
 
-      when(mockSaveMessageService.saveInboundMessage(any(), any())(any()))
-        .thenReturn(Future.successful(Right(())))
+      when(mockMovementMessageOrchestratorService.saveNCTSMessage(any(), any(), any())(any()))
+        .thenReturn(Future.successful(Right(InboundMessageRequest(arrival, GoodsReleased, GoodsReleasedResponse, message))))
 
       val application = baseApplicationBuilder
-        .overrides(bind[SaveMessageService].toInstance(mockSaveMessageService))
-        .overrides(bind[InboundRequestService].toInstance(mockInboundRequestService))
+        .overrides(bind[MovementMessageOrchestratorService].toInstance(mockMovementMessageOrchestratorService))
         .build()
 
       running(application) {
         val request = FakeRequest(POST, routes.NCTSMessageController.post(messageSender).url)
-          .withHeaders("channel" -> arrivalWithoutBox.channel.toString)
-          .withXmlBody(xml)
+          .withHeaders("channel" -> arrival.channel.toString)
+          .withXmlBody(<test></test>)
 
         val result = route(application, request).value
 
         status(result) mustEqual OK
-        header(LOCATION, result) mustBe Some(routes.MessagesController.getMessage(arrivalWithoutBox.arrivalId, arrivalWithoutBox.nextMessageId).url)
+        header(LOCATION, result) mustBe Some(routes.MessagesController.getMessage(arrival.arrivalId, arrival.nextMessageId).url)
       }
     }
 
-    "must return NotFound for an arrivalWithoutBox that does not exist" in {
+    "must return NotFound when the MovementMessageOrchestratorService returns an ArrivalNotFoundError" in {
```

Comment:
wasn't CLI-71 to return an OK in this scenario?

Reply:
We've since pivoted and this 404 to 200 mapping is now done by the `transit-movements-trader-router`